backend/prompt_studio/prompt_studio_core/prompt_studio_helper.py

```diff
@@ -785,6 +834,25 @@ def dynamic_indexer(
                 f"Error while indexing '{doc_name}'. {str(e)}"
             ) from e
 
+    @staticmethod
+    def wait_for_document_indexing(doc_id_key):
+        max_wait_time = settings.MAX_WAIT_TIME  # 30 minutes
+        wait_time = 0
+        polling_interval = settings.POLLING_INTERVAL  # Poll every 5 seconds
```

Comment:
[MINOR] I noticed there are a few libraries which take care of polling. Is there any reason for handling it ourself? Ref : https://pypi.org/project/polling/0.3.0/

Reply:
This is a simple polling method. I thought using a library would be overkill. Behind the scene libraries also have similar approach.